builtin/credential/approle/path_role.go

```diff
@@ -1314,12 +1400,21 @@ func (b *backend) pathRoleBoundCIDRListUpdate(ctx context.Context, req *logical.
 		return nil, nil
 	}
 
-	role.BoundCIDRList = data.Get("bound_cidr_list").([]string)
-	if len(role.BoundCIDRList) == 0 {
+	var cidrs []string
+	if cidrsIfc, ok := data.GetOk("bound_cidr_list"); ok {
+		cidrs = cidrsIfc.([]string)
+		role.BoundCIDRList = cidrs
```

Comment:
Can we prioritize `secret_id_bound_cidrs` as the one that gets checked before `bound_cidr_list`?

Also, should we really set the value into `role.BoundCIDRList`? Can we store it in `role.SecretIDBoundCIDRs` field instead? This can avoid an upgrade upon the role read.

Reply:
Ah! You just reminded me that I was also going to write a `fieldData.GetFirst()` method. Yes, I'll put the result directly into `role.SecretIDBoundCIDRs`.